src/languages/latex.js

```diff
@@ -1,70 +1,231 @@
-              ]
-            }
-          ]
-        }
+          {begin: KNOWN_CONTROL_WORDS},
+          {begin: KNOWN_CONTROL_SYMBOLS}
+        ]
+      },
+      {
+        endsParent: true,
+        variants: L3_VARIANTS
+      },
+      {
+        endsParent: true,
+        variants: DOUBLE_CARET_VARIANTS
+      },
+      {
+        endsParent: true,
+        relevance: 0,
+        variants: L2_VARIANTS
       }
     ]
   };
-
-  return {
-    name: 'LaTeX',
-    aliases: ['tex'],
-    contains: [
-      COMMAND,
-      {
-        className: 'formula',
-        contains: [COMMAND],
+  var ACTIVE_CHAR = {
+    className: 'keyword',
+    relevance: 0,
+    begin: /~|"/
+  };
+  var MACRO_PARAM = {
+    className: 'params',
+    relevance: 0,
+    begin: /#+\d?/
+  };
+  var DOUBLE_CARET_CHAR = {
+    variants: DOUBLE_CARET_VARIANTS
+  };
+  var SPECIAL_CATCODE = {
+    className: 'built_in',
+    relevance: 0,
+    begin: /[$&^_]/
+  };
+  var BRACES = {
+    className: 'built_in',
+    relevance: 0,
+    begin: /[{}]/
+  };
+  var MAGIC_COMMENT = hljs.COMMENT(
+    '% !TeX',
+    '$',
+    {
+      className: 'meta',
+      relevance: 10
+    }
+  );
+  var COMMENT = hljs.COMMENT(
+    '%',
+    '$',
+    {
+      relevance: 0
+    }
+  );
+  var EVERYTHING_BUT_BRACES_AND_VERBATIM = [
+    CONTROL_SEQUENCE,
+    ACTIVE_CHAR,
+    MACRO_PARAM,
+    DOUBLE_CARET_CHAR,
+    SPECIAL_CATCODE,
+    MAGIC_COMMENT,
+    COMMENT
+  ];
+  var EVERYTHING_BUT_VERBATIM = [...EVERYTHING_BUT_BRACES_AND_VERBATIM, BRACES];
+  var BRACE_GROUP_NO_VERBATIM = {
+    begin: /\{/, end: /\}/,
+    keywords: {
+      $pattern: /\{|\}/,
+      built_in: '{ }'
+    },
+    relevance: 0,
+    contains: ['self', ...EVERYTHING_BUT_BRACES_AND_VERBATIM]
+  }
+  var ARGUMENT_BRACES = hljs.inherit(
+    BRACE_GROUP_NO_VERBATIM,
+    {endsParent: true, contains: [BRACE_GROUP_NO_VERBATIM, ...EVERYTHING_BUT_BRACES_AND_VERBATIM]}
+  );
+  var ARGUMENT_BRACKETS = {
+    begin: /\[/, end: /\]/,
+    endsParent: true,
+    relevance: 0,
+    contains: [BRACE_GROUP_NO_VERBATIM, ...EVERYTHING_BUT_BRACES_AND_VERBATIM]
+  }
+  var ARGUMENT_ABSENT = {
+    begin: /(?=[.$])/, end: /(?=[.$])/,
+    endsParent: true,
+    relevance: 0
+  }
+  var SPACE_GOBBLER = {
+    begin: /\s+/,
+    endsParent: true,
+    relevance: 0
+  };
+  var ARGUMENT_M = [ARGUMENT_BRACES];
+  var ARGUMENT_O = [ARGUMENT_BRACKETS, ARGUMENT_ABSENT];
+  var ARGUMENT_NONE = [ARGUMENT_ABSENT];
+  var ARGUMENT_AND_THEN = function(arg, starts_mode) {
+    return {
+      contains: [SPACE_GOBBLER],
+      starts: {
         relevance: 0,
-        variants: [
-          {begin: /\$\$/, end: /\$\$/},
-          {begin: /\$/, end: /\$/}
-        ]
+        contains: arg,
+        starts: starts_mode
+      }
+    };
+  };
+  var CSNAME = function(csname, starts_mode) {
+    return hljs.inherit(
+      {
+        begin: '\\\\' + csname + '(?![a-zA-Z@:_])',
+        keywords: {$pattern: /\S+/, keyword: '\\' + csname}
```

Comment:
Pretty sure `\S+` if FAR too broad... can we pin that down to what is actually a valid identifier?

Reply:
Of course, `\\[a-zA-Z]+` is sufficient here. I just didn't think it made a difference, since this regex is only applied to the one string on the same line. Will change that.